python/examples/core/harness.py

```diff
@@ -286,7 +297,7 @@ def test_harness(
             transform=expert,
             dump_ir_to_file=kwargs.get('dump_ir_to_file', ''))
 
-        problem.run(
+        results['expert'][expert] = problem.run(
```

Comment:
What about flattening the dictionary and using something like `expert.__name__` ? They names aren't very human-friendly, but that is one of the feature requests on them.

Reply:
Good idea, I've used `str(expert)` as the key here which is more unique (includes a heap address hash) than just a type name.